cl/lib/email_backends.py

```diff
@@ -18,6 +21,93 @@
 )
 
 
+def incr_email_temp_counter(r: Redis) -> None:
+    """increments the temporary counter and adds a new
+    element to the sorted set once it reaches the value of
+    the EMAILS_TEMP_COUNTER setting.
+
+    Args:
+        r (Redis): The Redis DB connection interface.
+    """
+    temp_counter = r.get("email:temp_counter")
+
+    if temp_counter:
+        if int(temp_counter) + 1 >= settings.EMAIL_MAX_TEMP_COUNTER:
+            current_time = time.time_ns()
+            pipe = r.pipeline()
+            pipe.zadd(
+                "email:delivery_attempts", {str(current_time): current_time}
+            )
+            pipe.set("email:temp_counter", 0)
+            pipe.execute()
+            return
+
+    pipe = r.pipeline()
+    pipe.incr("email:temp_counter")
+    pipe.execute()
+    return
+
+
+def get_attempts_in_window(r: Redis) -> int:
+    """
+    Returns the number of elements stored in the set. This method
+    check the current time, and shave off any attempts in the sorted set
+    that are outside of our window, then check the cardinality of the
+    sorted set
+
+    Args:
+        r (Redis): The Redis DB connection interface.
+
+    Returns:
+        int: number of elements stored in the set
+    """
+    current_time = time.time_ns()
+    trim_time = current_time - (24 * 60 * 60 * 1_000_000_000)
+    pipe = r.pipeline()
+    # Removes attempts outside the current window
+    pipe.zremrangebyscore("email:delivery_attempts", 0, trim_time)
+    # Get number of elements in the set
+    pipe.zcard("email:delivery_attempts")
+    _, size = pipe.execute()
+    return int(size)
+
+
+def get_email_count(r: Redis) -> int:
+    """Returns the number of email sent in the last 24 hours.
+
+    Args:
+        r (Redis): The Redis DB connection interface.
+
+    Returns:
+        int: number of emails sent in the last 24 hours
+    """
+    temp_counter = r.get("email:temp_counter")
+    previous_attempts = get_attempts_in_window(r)
+    if not temp_counter:
+        return previous_attempts * settings.EMAIL_MAX_TEMP_COUNTER
+    return (
+        int(temp_counter) + previous_attempts * settings.EMAIL_MAX_TEMP_COUNTER
```

Comment:
I think this is the same math, but it's more clear. It both reads and operates left to right and it echos the code in `if not temp_counter`, above:

```suggestion
        previous_attempts * settings.EMAIL_MAX_TEMP_COUNTER + int(temp_counter)
```